File: codewars/8kyu/Convert a Number to a String!/main.go
package kata

import (
	"bytes"
	"math"
)

func NumberToString(n int) string {
	if n == 0 {
		return "0"
	}

	isNegative := n < 0
	if isNegative {
		n = -n
	}

	var buffer bytes.Buffer
	for n > 0 {
		digit := n % 10
		buffer.WriteByte(byte(digit) + '0')
		n = int(math.Floor(float64(n) / 10.0))
	}

	if isNegative {
		buffer.WriteByte('-')
	}

	inner := buffer.Bytes()

	for i, j := 0, len(inner)-1; i < j; i, j = i+1, j-1 {
		inner[i], inner[j] = inner[j], inner[i]
	}

	return string(inner)
}

File: codewars/8kyu/Convert a Number to a String!/main.rs
fn number_to_string(i: i32) -> String {
    let mut n = i;

    if n == 0 {
        return "0".to_string();
    }

    let mut is_negative = false;

    if n < 0 {
        is_negative = true;
        n = -n
    }

    let mut digits = Vec::new();

    while n > 0 {
        let digit = n % 10;
        digits.push(digit as u8 + b'0');
        n /= 10;
    }

    if is_negative {
        digits.push(b'-');
    }

    digits.reverse();

    String::from_utf8(digits).unwrap()
}
